File: src/test/java/com/integreety/yatspec/e2e/captor/http/mapper/PropertyServiceNameDeriverShould.java
package com.integreety.yatspec.e2e.captor.http.mapper;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class PropertyServiceNameDeriverShould {

    @ParameterizedTest
    @CsvSource(value = {
            "Global User Service, GlobalUser",
            "User Address, UserAddress"
    })
    public void deriveServiceName(final String appName, final String expectedServiceName) {
        final PropertyServiceNameDeriver underTest = new PropertyServiceNameDeriver(appName);
        assertThat(underTest.getServiceName(), is(expectedServiceName));
    }
}

File: src/test/kotlin/io/lsdconsulting/lsd/distributed/interceptor/captor/http/derive/PropertyServiceNameDeriverShould.kt
package io.lsdconsulting.lsd.distributed.interceptor.captor.http.derive

import io.lsdconsulting.lsd.distributed.interceptor.captor.common.PropertyServiceNameDeriver
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class PropertyServiceNameDeriverShould {

    @ParameterizedTest
    @CsvSource(value = ["Global User Service, GlobalUser", "User Address, UserAddress"])
    fun deriveServiceName(appName: String, expectedServiceName: String) {
        val underTest = PropertyServiceNameDeriver(appName)
        assertThat(underTest.serviceName, `is`(expectedServiceName))
    }
}
